src/datapump/sync/fire_alerts.py

```diff
@@ -30,90 +27,77 @@
 
 
 def process_active_fire_alerts(alert_type):
+    nrt_s3_directory = f"nasa_{alert_type.lower()}_fire_alerts/{VERSIONS[alert_type]}/vector/epsg-4326/tsv/near_real_time"
+    last_saved_date, last_saved_min = _get_last_saved_alert_time(nrt_s3_directory)
+    LOGGER.info(f"Last saved row datetime: {last_saved_date} {last_saved_min}")
+
     LOGGER.info(f"Retrieving fire alerts for {alert_type}")
-    response = requests.get(ACTIVE_FIRE_ALERTS_48HR_CSV_URLS[alert_type])
+    rows = []
+    with shapefile.Reader(ACTIVE_FIRE_ALERTS_7D_SHAPEFILE_URLS[alert_type]) as sf:
+        LOGGER.info(f"Shapefile has {len(sf)} records")
+        for shape_record in sf.iterShapeRecords():
+            row = shape_record.record.as_dict()
+            row["ACQ_DATE"] = row["ACQ_DATE"].strftime("%Y-%m-%d")
+            row["LATITUDE"] = shape_record.shape.points[0][1]
+            row["LONGITUDE"] = shape_record.shape.points[0][0]
+            # For VIIRS we only want first letter of confidence category,
+            # to make NRT category same as scientific
+            if alert_type == "viirs":
+                row["CONFIDENCE"] = row["CONFIDENCE"][0]
+
+            if row["ACQ_DATE"] > last_saved_date or (
+                row["ACQ_DATE"] == last_saved_date and
+                row["ACQ_TIME"] > last_saved_min
+            ):
+                rows.append(row)
 
-    if response.status_code != 200:
+    if not rows:
         raise Exception(
-            f"Unable to get active {alert_type} fire alerts, FIRMS returned status code {response.status_code}"
+            f"{alert_type} shapefile contained no new records since {last_saved_date} {last_saved_min}"
```

Comment:
I guess this will just leave a reasonable slack messages?  If the message is quite big/complicated, we may only  want to do this if we haven't seen new records in 3-7 days?

Reply:
I don't think it's too big, but I'll test in staging!